### ISSUE TITLE
Document which architectures this crate is supposed to work on and/or provide fallbacks

### ISSUE THREAD

**Issue by infinity0** [hsivonen/simd#25]
e.g. fails on Debian ppc64 https://buildd.debian.org/status/fetch.php?pkg=rust-simd&arch=ppc64&ver=0.2.2-1&stamp=1536046860&raw=0

However encoding-rs unconditionally depends on this crate, which means it can't be built on ppc64. Either this crate should provide fallbacks so it can be built, or your dependents should add the appropriate conditionals to their Cargo.toml, which you should document.

**Comment by infinity0** [comment]
Ideally this crate should build with fallbacks on unsupported architectures, since this places the least amount of burden on other people.

**Comment by luser** [comment]
> However encoding-rs unconditionally depends on this crate, which means it can't be built on ppc64.

I don't think this has ever been true, has it? encoding-rs hides it behind a `simd-accel` feature:
https://github.com/hsivonen/encoding_rs/blob/5052145b1ffd19f4abb00006b9c2fb3eebb28ec0/Cargo.toml#L25

We enable that by way of a `--enable-rust-simd` option in the Firefox build.

**Comment by infinity0** [comment]
> encoding-rs hides it behind a simd-accel feature:

By "unconditionally" I meant it is not conditioned on any specific architectures. As a consumer of the encoding-rs crate who is ignorant of which architectures have which accelerations, I can't just blanket enable the `simd-accel` feature for all architectures.

**Comment by mzpqnxow** [comment]
2 years later, I'm adding my $0.02 to this. I'm also stuck on a POWER8 machine, unable to use some relatively basic packages because they rely on encoding_rs, which in turn relies on simd. With simd having neither explicit ppc64le support nor a generic fallback, I'm hosed, unless I can go convince the encoding_rs maintainers to provide fallback on their side- which may be OK, but I'm wondering how many other packages will have this same issue.

Is this package still maintained? I see this issue has been open for quite a long time

Thanks
